site/_posts/2020-09-21-troubleshooting-containerized-oidc.md

```diff
@@ -0,0 +1,317 @@
+which is implemented using the [mod_auth_openidc](https://github.com/zmartzone/mod_auth_openidc/blob/master/README.md)
+Apache module.
+
+The HTTPD pod is used to configure Apache's mod_auth_openidc module.
+
+Therefore the httpd pod's log file may contain errors related to mod_auth_openidc configuration.
+
+The orchestrator pod will log any errors encountered creating the other pods in the ManageIQ deployment.
+
+The following example shows how to check the httpd pod's logs.
+
+  ```bash
+  $ oc login -u admin -p <pw> https://<my-container-host>/
+  $ oc logs httpd-abcdef123-12345 -n my-miq-project
+  ```
+
+## OpenID-Connect Provider Accessible
+---------------------------------------------------------------------
+
+**Question** Can user data be obtained from the OpenID-Connect Provider?
+
+ManageIQ has a REST API. There is a blog post that describes how to use the ManageIQ REST API
+when configured with OpenID-Connect. 
+see: [Using the API with OpenID-Connect Authentication](https://www.manageiq.org/blog/2020/09/oidc-auth-api-usage/)
+
+One of the steps described in this blog post in the **Example: Using a JWT Token** contains an optional
+step to do an introspection to the OpenID-Connect provider for a specific user.
+
+The following snippet from the [Using the API with OpenID-Connect Authentication](https://www.manageiq.org/blog/2020/09/oidc-auth-api-usage/) blog post shows the commands that will
+directly query the OpenID-Connect Provider for a specific user.
+
+  ```bash
+  # Accessing the ManageIQ API with the access_token from a JWT Token
+
+  # Step 1: Request the JWT Token from the OpenID-Connect provider
+  jwt_token=$(curl -k -L --user ${user}:${password} -X POST -H "Content-Type: application/x-www-form-urlencoded" -d "grant_type=password" -d "client_id=${oidc_client_id}" -d "client_secret=${oidc_client_secret}" -d "username=${user}" -d "password=${password}" ${oidc_provider_token_endpoint} )
+
+  # Step 2: Retrieve the access_token from the JWT
+  access_token=$(echo $jwt_token | jq -r '.access_token')
+
+  # Step 3 (OPTIONAL) Use the access_token to do an introspection to get the details for the user associated with the JWT
+  curl -k -L --user ${oidc_client_id}:${oidc_client_secret} -X POST -H "Content-Type: application/x-www-form-urlencoded" -d "token=${access_token}" ${oidc_auth_introspection_endpoint}
+  ```
+
+The output from the above introspection *curl* command should look similar to this:
+
+  ```json
+  "name": "Bob Youruncle",
+  "given_name": "Bob",
+  "family_name": "Youruncle",
+  "preferred_username": "byouruncle",
+  "email": "byouruncle@example.com",
+  ...
+  "domain": "byouruncle.example.com",
+  "groups": [
+    "EvmGroup-super_administrator"
+  ],
+  "username": "byouruncle",
+  ```
+
+If an issue is encountered where a specific user is unable to log in it would be valuable to confirm this introspection returns the expected data, when performed for the failing user.
```

Comment:
👍 thanks for adding this, this will be helpful.

Reply:
@abellotti , @gtanzillo  and @bdunne Thank you for you help reviewing this!